assets/app/scripts/controllers/services.js

```diff
@@ -24,7 +25,35 @@ angular.module('openshiftConsole')
       $scope.emptyMessage = "No services to show";
       updateFilterWarning();
       console.log("services (subscribe)", $scope.unfilteredServices);
-    }));    
+    }));
+
+    watches.push(DataService.watch("routes", $scope, function(routes){
+        $scope.routesByService = routesByService(routes.by("metadata.name"));
+        console.log("routes (subscribe)", $scope.routesByService);
+    }));
+
+    var routesByService = function(routes) {
+        var routeMap = {};
+        angular.forEach(routes, function(route){
+            routeMap[route.serviceName] = routeMap[route.serviceName] || {};
+            route.url = getRouteUrl(route);
```

Comment:
we try really hard not to mutate the objects we get back from the API, so that anyone who knows the API can work with the objects in the UI automatically. Maybe make the routeURL function a filter so it can be used in the view (and name it more descriptively, like `routeWebURL`)?